lib/views.js

```diff
@@ -5,163 +5,58 @@
-
-        if (form) {
-          emit([centerId, form, doc.reported_date], 1);
-          emit([valid, centerId, form, doc.reported_date], 1);
-
-          emit([centerId, '*', doc.reported_date], 1);
-          emit([valid, centerId, '*', doc.reported_date], 1);
-        } else {
-          emit([centerId, 'null_form', doc.reported_date], 1);
-          emit([valid, centerId, 'null_form', doc.reported_date], 1);
-        }
-      }
-      if (districtId) {
-        emit([districtId, doc.reported_date], 1);
-        emit([valid, districtId, doc.reported_date], 1);
-
-        if (form) {
-          emit([districtId, form, doc.reported_date], 1);
-          emit([valid, districtId, form, doc.reported_date], 1);
-
-          emit([districtId, '*', doc.reported_date], 1);
-          emit([valid, districtId, '*', doc.reported_date], 1);
-        } else {
-          emit([districtId, 'null_form', doc.reported_date], 1);
-          emit([valid, districtId, 'null_form', doc.reported_date], 1);
-        }
-      }
-      if (clinicId && districtId) {
-        emit([districtId, clinicId, doc.reported_date], 1);
-        emit([valid, districtId, clinicId, doc.reported_date], 1);
-
-        if (form) {
-          emit([districtId, clinicId, form, doc.reported_date], 1);
-          emit([valid, districtId, clinicId, form, doc.reported_date], 1);
-
-          emit([districtId, clinicId, '*', doc.reported_date], 1);
-          emit([valid, districtId, clinicId, '*', doc.reported_date], 1);
-        } else {
-          emit([districtId, clinicId, 'null_form', doc.reported_date], 1);
-          emit([valid, districtId, clinicId, 'null_form', doc.reported_date], 1);
-        }
+        emit([valid, form, doc.reported_date]);
+        emit([valid, '*', doc.reported_date]);
       }
-      if (centerId && districtId) {
-        emit([districtId, centerId, doc.reported_date], 1);
-        emit([valid, districtId, centerId, doc.reported_date], 1);
 
-        if (form) {
-          emit([districtId, centerId, form, doc.reported_date], 1);
-          emit([valid, districtId, centerId, form, doc.reported_date], 1);
-
-          emit([districtId, centerId, '*', doc.reported_date], 1);
-          emit([valid, districtId, centerId, '*', doc.reported_date], 1);
-        } else {
-          emit([districtId, centerId, 'null_form', doc.reported_date], 1);
-          emit([valid, districtId, centerId, 'null_form', doc.reported_date], 1);
+      var parent = doc.contact;
+      while(parent) {
+        if (parent._id) {
+          emit([parent._id, doc.reported_date]);
+          if (form) {
+            emit([valid, parent._id, form, doc.reported_date]);
+            emit([valid, parent._id, '*', doc.reported_date]);
+          }
         }
+        parent = parent.parent;
       }
     }
   }
 };
 
 // Only used in ./scripts/delete_training_data_utils.js
-exports.data_records_by_district = {
+exports.data_records_by_ancestor = {
```

Comment:
IMO `parent` is the direct ancestor, `ancestor` is the parent, grandparent, etc. This means if you're deleting a district it'll delete all data_records in any clinic in any health center in that district. I think this is what's intended in that script.